>From 456b5f5c8b1a9a85615054673e70ad74f30da4e9 Mon Sep 17 00:00:00 2001
From: Cyril Brulebois <kibi@debian.org>
Date: Wed, 25 May 2016 17:54:58 +0200
Subject: [PATCH] Add support for retrieving and listing Contents-udeb-* files.

Since apt-file now relies on apt-get indextargets to figure out which Contents
files to look for, it's important for apt to downloads Contents-udeb-* instead
of trying to download non-existing debian-installer/Contents-* files.

Rewrite MetaKey and ShortDesc accordingly, which enables apt-file to search into
udebs.

Signed-off-by: Cyril Brulebois <kibi@debian.org>
---
 apt-pkg/deb/debmetaindex.cc | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/apt-pkg/deb/debmetaindex.cc b/apt-pkg/deb/debmetaindex.cc
index f756cdb..4184d00 100644
--- a/apt-pkg/deb/debmetaindex.cc
+++ b/apt-pkg/deb/debmetaindex.cc
@@ -238,6 +238,16 @@ static void GetIndexTargetsFor(char const * const Type, std::string const &URI,
 		     ShortDesc = SubstVar(ShortDesc, varname, O->second);
 		     LongDesc = SubstVar(LongDesc, varname, O->second);
 		     Identifier = SubstVar(Identifier, varname, O->second);
+
+		     // Rewrite debian-installer/Contents into Contents-udeb for
+		     // MetaKey and Contents into Contents-udeb for ShortDesc:
+                     std::string di_contents_token = "debian-installer/Contents-";
+                     std::string contents_token = "Contents-";
+                     size_t di_pos = MetaKey.find(di_contents_token);
+                     if (di_pos != std::string::npos) {
+                       MetaKey = MetaKey.substr(0, di_pos) + "Contents-udeb-" + MetaKey.substr(di_pos + di_contents_token.length());
+                       ShortDesc = "Contents-udeb-" + ShortDesc.substr(contents_token.length());
+                     }
 		  }
 
 		  {
-- 
2.8.1

